health/health_log.c

```diff
@@ -174,6 +176,27 @@ inline ssize_t health_alarm_log_read(RRDHOST *host, FILE *fp, const char *filena
                 continue;
             }
 
+            // Check if we got last_repeat field
+            time_t last_repeat = 0;
+            if(entries > 27) {
+                char* alarm_name = pointers[13];
+                last_repeat = (time_t)strtoul(pointers[27], NULL, 16);
+
+                RRDCALC *rc = NULL;
```

Comment:
I don't like this part here, it seems to be a leftover from the previous code. I realize it's difficult to go from the alarm entry in the log to the rrdcalc after a restart of netdata, but we need to solve this problem, also so that we can center the chart for an inactive alarm. You should find a way to use that avl_tree to get to the rrdcalc entry efficiently.

Reply:
This is fixed now, I will work in the last request for we finish this!